ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -105,15 +104,16 @@ def process_result(
         # policy breaks and matches are modified in the functions leak_dictionary_by_ignore_sha and censor_content.
         # Previously process_result was executed only once, so it did not create any issue.
         # In the future we could rework those functions such that they do not change what is in the result.
-        policy_breaks = deepcopy(result.scan.policy_breaks)
+        result = deepcopy(_result)
```

Comment:
I think you can overwrite `result` with a deepcopy of itself. This way you don't need to rename `result` to `_result` in the arguments (which looks a bit odd, and could break if using named arguments)

Reply:
I changed the line anyway because in some cases it was not possible to make a deep copy of result but I'm not sure my current version truly works :confused: